pkg/versiongetter/cargo.go

```diff
@@ -23,11 +24,11 @@ func NewCargo(client CargoClient) *CargoVersionGetter {
 	}
 }
 
-func (c *CargoVersionGetter) Get(ctx context.Context, pkg *registry.PackageInfo, filters []*Filter) (string, error) {
+func (c *CargoVersionGetter) Get(ctx context.Context, _ *logrus.Entry, pkg *registry.PackageInfo, filters []*Filter) (string, error) {
```

Comment:
There are five classes/structs that implement the interface [`VersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/version_getter.go#L10):

1. [`GeneralVersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/general.go#L12)
2. [`CargoVersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/cargo.go#L16)
3. [`GitHubReleaseVersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/github_release.go#L18)
4. [`GitHubTagVersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/github_tag.go#L12)
5. [`MockVersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/mock_version_getter.go)

Since I’ve modified the [`VersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/version_getter.go#L10). Removing the  `_ *logrus.Entry,` will result in a compile error in [`GeneralVersionGetter.get`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/general.go#L42).



After considering the feature of log enhancement, I think there are two options:

1. Keep it simple. Do not modify the interface  [`VersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/version_getter.go#L10), log the ERROR in  ‎[`FuzzyGetter.Get`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/fuzzy_getter.go#L29)‎ only.
2. Modify the interface [`VersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/version_getter.go#L10) and log the GH API Rate Limit info as DEBUG level. The [`CargoVersionGetter`](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/cargo.go#L16) has to be modified.

What do you think about it? If you have a better approach, please tell me.

Reply:
> Since I’ve modified the [VersionGetter](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/version_getter.go#L10). Removing the _ *logrus.Entry, will result in a compile error in [GeneralVersionGetter.get](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/general.go#L42).

Oh, I see. Thank you for your explanation.

> Keep it simple. Do not modify the interface [VersionGetter](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/version_getter.go#L10), log the ERROR in ‎[FuzzyGetter.Get](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/fuzzy_getter.go#L29)‎ only.

I disagree with this option. We usually pass *logrus.Entry as an argument to take over the context (log fields).

> Modify the interface [VersionGetter](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/version_getter.go#L10) and log the GH API Rate Limit info as DEBUG level. The [CargoVersionGetter](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/cargo.go#L16) has to be modified.

The log about GitHub API rate limit should be outputted only when GitHub API rate limit error occurs.

https://pkg.go.dev/github.com/google/go-github/github#hdr-Rate_Limiting

Reply:
> The log about GitHub API rate limit should be outputted only when GitHub API rate limit error occurs.
> 
> https://pkg.go.dev/github.com/google/go-github/github#hdr-Rate_Limiting

Ah, sorry. Let me correct.
Debug log is fine.